### PR TITLE
Fix non 3d annotations appearing in 3D viewport

### PR DESCRIPTION
**Pull request by rodrigobasilio2022**
#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

- [] I have run the `yarn build:update-api` to update the API documentation, and have
  committed the changes to this PR. (Read more here https://www.cornerstonejs.org/docs/contribute/update-api)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] "OS: 
- [] "Node version: 
- [] "Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

### CODE REVIEW

File: packages/tools/src/utilities/planar/filterAnnotationsForDisplay.ts [hunk 1]
```diff
@@ -71,7 +71,16 @@ export default function filterAnnotationsForDisplay(
       spacingInNormalDirection
     );
   } else if (viewport instanceof VolumeViewport3D) {
-    return annotations;
+    return annotations.filter((annotation) => {
+      if (!annotation.isVisible) {
+        return false;
+      }
+
+      if (annotation.metadata.toolName !== 'Length3D') {
```
Comment (ladeirarodolfo) [inline packages/tools/src/utilities/planar/filterAnnotationsForDisplay.ts:79]: The solution has to be generic.

For that create a property on tools and check it here. Something like
return metadata.toolContext !== '3D'

File: packages/tools/src/utilities/planar/filterAnnotationsForDisplay.ts [hunk 2]
```diff
@@ -71,7 +71,16 @@ export default function filterAnnotationsForDisplay(
       spacingInNormalDirection
     );
   } else if (viewport instanceof VolumeViewport3D) {
-    return annotations;
+    return annotations.filter((annotation) => {
+      if (!annotation.isVisible) {
```
Comment (ladeirarodolfo) [inline packages/tools/src/utilities/planar/filterAnnotationsForDisplay.ts:75]: I believe the check for visible or not tools are done already in other place, no?

**Review by ladeirarodolfo — CHANGES_REQUESTED** [review]
Few comments only. Thanks Rodrigo

File: packages/tools/src/utilities/planar/filterAnnotationsForDisplay.ts (hunk 1 above)
Comment (rodrigobasilio2022) [reply packages/tools/src/utilities/planar/filterAnnotationsForDisplay.ts]: done

File: packages/tools/src/utilities/planar/filterAnnotationsForDisplay.ts (hunk 2 above)
Comment (rodrigobasilio2022) [reply packages/tools/src/utilities/planar/filterAnnotationsForDisplay.ts:75]: Here we filter to not habve to handle it in renderAnnotation. I dont see a problem. We could maintain both for code safety

Comment (ladeirarodolfo) [reply packages/tools/src/utilities/planar/filterAnnotationsForDisplay.ts:75]: I would agree with you, but less change in this layer better will be

File: packages/tools/src/types/AnnotationTypes.ts [hunk 3]
```diff
@@ -47,6 +47,10 @@ type Annotation = {
      * VolumeId of the volume that the tool was configured to work on.
      */
     volumeId?: string;
+    /**
+     * The context of the annotation: 2D or 3D
```
Comment (ladeirarodolfo) [inline packages/tools/src/types/AnnotationTypes.ts:51]: Add a comment saying if empty it will be treated as 2d

**Review by ladeirarodolfo — CHANGES_REQUESTED** [review]

File: packages/tools/src/utilities/planar/filterAnnotationsForDisplay.ts [hunk 4]
```diff
@@ -71,7 +71,16 @@ export default function filterAnnotationsForDisplay(
...
+        return false;
+      }
+
+      if (annotation.metadata?.toolContext !== '3D') {
```
Comment (ladeirarodolfo) [inline packages/tools/src/utilities/planar/filterAnnotationsForDisplay.ts:79]: return (annotation.metadata?.toolContext?.toUpperCase()) === '3D', no?

(less code)

**Review by ladeirarodolfo — CHANGES_REQUESTED** [review]

**Review by ladeirarodolfo — APPROVED** [review]

**Pull request merged**